### ISSUE TITLE
Popups don't respond to content changes

### ISSUE THREAD

**Issue by daviestar** [alex3165/react-mapbox-gl#31]
Hello, been using this binding for a few days now and have to say, it's fantastic!!!! I have ~50,000 points/polylines displayed and performance is excellent, especially compared to the alternative. 👍 

I have a few basic click-handlers inside popup content which toggle classes (tabs).  I've found changes within popup content do try to trigger a re-render ([`componentWillReceiveProps`](https://github.com/alex3165/react-mapbox-gl/blob/master/src/popup.js#L60-L83) is called) but the content is not updated.

Here's the quick fix I made:
```javascript
componentWillReceiveProps(nextProps) {
  const { div, popup } = this;
  const {
    children,
    coordinates,
    dangerouslySetInnerHTML,
    text
  } = nextProps;

  if (children) {
    render(children, div);
  } else if (
    this.props.dangerouslySetInnerHTML &&
    dangerouslySetInnerHTML !== this.props.dangerouslySetInnerHTML
  ) {
    popup.setHTML(dangerouslySetInnerHTML);
  } else if (text !== this.props.text) {
    popup.setText(text);
  }

  if (this.props.coordinates !== coordinates) {
    popup.setLngLat(coordinates);
  }
}
```

**Comment by alex3165** [comment]
Hello, thank you, can you open a PR ?

**Comment by daviestar** [comment]
sure, although not sure when as I'm snowed under at the moment. cheers, dave

**Comment by infacq** [comment]
+1

**Comment by alex3165** [comment]
Fixed and published with a new version 0.9.0.
https://github.com/alex3165/react-mapbox-gl/releases/tag/v0.9.0
